### PR TITLE
CPack: Debian package

### PR DESCRIPTION
**Pull request by ctrlaltca**
This PR makes the needed modification in order to build a working, redistributable debian (ubuntu) package.
There are three main changes:
 * CMake/CPack directives to produce a working package;
 * CMake changes to create an debian-compatible version number (reference: http://debian.2.n7.nabble.com/Git-Package-Versioning-td1390729.html);
 * Translation path was hardcoded in linux; detect it at runtime instead.

Once this will get in, the following instructions can be used to create a package from ubuntu:
```
# Ubuntu 14.04 LTS (Trusty Tahr)

# Update packages list
sudo apt-get update
# Install the needed prerequisites
sudo apt-get install -y git build-essential g++ cmake \
	libprotobuf-dev protobuf-compiler \
	qt5-default qttools5-dev qttools5-dev-tools qtmultimedia5-dev libqt5multimedia5-plugins libqt5svg5-dev libqt5sql5-mysql
# Get a copy of Cockatrice's source code
git clone git://github.com/Cockatrice/Cockatrice
# Create a directory to host the build process
cd Cockatrice
mkdir build
cd build
# Configure the build
cmake .. -DWITH_SERVER=1
# Compile Cockatrice and build a debian package
make package -j2
# Install the package
sudo dpkg -i Cockatrice*.deb
# Hotkeys are currently broken for Qt5 apps on ubuntu (see bug: https://bugs.launchpad.net/appmenu-qt5/+bug/1380702)
# The only current known workaround is to remove the problematic package 
sudo apt-get remove appmenu-qt5
```

### CODE REVIEW

**Comment by tooomm** [comment]
this will replace #491 ?

**Comment by ctrlaltca** [comment]
The target for both PRs is to get a deb package created. Imho using cmake's builting capabilities is a more maintainable way to accomplish the same result.

**Comment by Daenyth** [comment]
Will this work on Ubuntu 12.05? (Latest LTS, travis runs it)

File: travis-compile.sh [hunk 1]
```diff
@@ -10,6 +10,7 @@ if [[ $TRAVIS_OS_NAME == "osx" && $QT4 == 0 ]]; then
 fi
 if [[ $TRAVIS_OS_NAME == "linux" && $QT4 == 0 ]]; then
   prefix="-DCMAKE_PREFIX_PATH=`echo /opt/qt5*/lib/cmake/`"
+  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:`echo /opt/qt5*/lib/`
```
Comment (Daenyth) [inline travis-compile.sh]: Can you do `export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$(echo /opt/qt5*/lib/)`

**Comment by ctrlaltca** [comment]
From the job log on travis: https://travis-ci.org/Cockatrice/Cockatrice/jobs/85201803
```
[100%] Built target cockatrice
Run CPack packaging tool...
CPack: Create package using DEB
CPack: Install projects
CPack: - Run preinstall target for: Cockatrice
CPack: - Install project: Cockatrice
CPack: Create package
CPackDeb: - Generating dependency list
CPack: - package: /home/travis/build/Cockatrice/Cockatrice/build/Cockatrice-0.0.1~git20151013.f1fa9c6.deb generated.
```
The only problem is that this package will be dependent to the qt packages that we manually install on travis from ppa:beineri/opt-qt541

File: travis-compile.sh (hunk 1 above)
Comment (ZeldaZach) [inline travis-compile.sh]: Any reason @Daenyth ? The `` works as a bash phrase and it's essentially the same thing (Just curious on your reasoning)

Comment (ctrlaltca) [inline travis-compile.sh]: I did, but it broke the build.

Comment (Daenyth) [inline travis-compile.sh]: @ZeldaZach It's better style in general. Backticks should only be used when you need POSIX sh-compatibility.  And the quoting protects against spaces in the path (though it doesn't matter in this case)

**Comment by Daenyth** [comment]
@ctrlaltca Looks like there's an issue with Qt4 cmake setup now.

**Comment by ctrlaltca** [comment]
> @ctrlaltca Looks like there's an issue with Qt4 cmake setup now on linux.
Nope, it's on builds where ```QT4=0```, aka Qt5 builds. It's due to apt-get not being able to download qt5 from the ppa.. i've tried to re-launch the failed build jobs.

**Comment by ctrlaltca** [comment]
Update: the beineri ppa has been updated to qt 5.4.2, and 5.4.1 packages have been removed, thus the failure. I've updated the travis script in order to get 5.4.2.

**Comment by woogerboy21** [comment]
```
By installing you accept licenses for the packages.
nsis not installed. An error occurred during installation:
 The remote server returned an error: (500) Internal Server Error.
The install of nsis was NOT successful.
nsis not installed. An error occurred during installation:
 The remote server returned an error: (500) Internal Server Error.
 
Chocolatey installed 0/1 package(s). 1 package(s) failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
Failures:
 - nsis
```
Derp...

**Comment by Daenyth** [comment]
I bumped the build. It's running now.

**Comment by ctrlaltca** [comment]
Yep, sometimes also appveyor fails at downloading dependencies.

**Comment by Daenyth** [comment]
I'd say successful CI builds are reason enough for :shipit:

Second?

**Comment by Daenyth** [comment]
@ZeldaZach Do you want to have a look at the build results?

**Comment by ZeldaZach** [comment]
I'm happy with it. :shipit:

**Pull request merged**